### DISCUSSION TITLE
No output training custom model locally

### DISCUSSION THREAD

**Discussion by sstainba** [ultralytics/yolov5#8733] [Q&A]
I have a dataset of about 300 images all 1280x1024 where I have a single class labeled. I was able to successfully train with the YoloV5/Roboflow notebook on Collab.  I am trying to train locally on my 3060 w/ 12GB so I set the batch size to 4 and 5 epochs to start.  However, if I have my batch size as anything greater than 1, I get output files in the runs/train folder.  My GPU stays at about 12% or so and only uses like 3GB of memory.  I've waited like this for about 15 minutes before killing the process.  Can someone please tell me what I might be doing wrong?  

This is what my train command looks like:
!python train.py --rect --img 1024 --imgsz 1280 --batch 8 --epochs 5 --data {dataset.location}/data.yaml --name {OUTPUT_NAME} --cache

If I set the batch = 1, I get some actual output.

**Comment by glenn-jocher** [comment]
@sstainba it appears you may have environment problems. Please ensure you meet all dependency requirements if you are attempting to run YOLOv5 locally.  If in doubt, create a new virtual [Python 3.9](https://www.python.org/) environment, clone the latest repo ([code changes daily](https://github.com/ultralytics/yolov5/commits/master)), and `pip install` [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) again from scratch.

💡 ProTip! Try one of our verified environments below if you are having trouble with your local environment.


## Requirements

[**Python>=3.7.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.7**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

[Models](https://github.com/ultralytics/yolov5/tree/master/models) and [datasets](https://github.com/ultralytics/yolov5/tree/master/data) download automatically from the latest YOLOv5 [release](https://github.com/ultralytics/yolov5/releases) when first requested.

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/GCP-Quickstart)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/AWS-Quickstart)
- **Docker Image**. See [Docker Quickstart Guide](https://github.com/ultralytics/yolov5/wiki/Docker-Quickstart) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), validation ([val.py](https://github.com/ultralytics/yolov5/blob/master/val.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by sstainba** [reply]
I am using a virtual environment with Python 3.9.2.  I'm using CUDA 11.6 with the corresponding CuDNN.  I am running the Jupyter notebook provided by you guys in the custom training tutorial (using Roboflow).  The only thing I've changed is the paths.  And to be clear, the training runs, but ONLY if the batch size is 1.  Any other value causes it to idle with no output.  

Is this possibly because my CUDA version is too new?
Also, the tutorials all provide the batch flag as "--batch" but in the train.py script it seems the flag is named "--batch-size".  I'm not a Python dev, but is that an issue or no?

**Reply by glenn-jocher** [reply]
Official YOLOv5 notebook is here, that that.
https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb?hl=en

argument names can be shortened in python

**Reply by sstainba** [reply]
I'm using the notebook provided at https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data in the tutorial for Training Custom Data.  Is that not valid?

I've also matched my CUDA/CuDNN version to that of Collab (11.3) but I'm still having the same problem.  Is there more information you can provide?

**Reply by glenn-jocher** [reply]
@sstainba you can check the output of AMP checks on train start, there should be a green checkmark there to verify that your CUDA/torch versions are compatible and can produce correct trianing.
